### ISSUE TITLE
Multiple HTML attributes not editable on Categories

### ISSUE THREAD

**Issue by adamlavery** [magento/magento2#38843]
### Preconditions and environment

CE 2.4.7

Needs a custom category text attribute to be added that uses the WYSIWYG editor. See the web for example of how to do that.

This arose after upgrading from 2.4.3 to 2.4.7. Upgrade went through 2.4.4, 2.4.5 & 2.4.6 so no idea what version this bug appeared in.

### Steps to reproduce

1. Add a custom text category attribute e.g. https://developer.adobe.com/commerce/frontend-core/ui-components/howto/add-category-attribute/

2. Add to the category edit page as wysiwyg i.e. 

```
        <field name="custom_description" template="ui/form/field" sortOrder="100" formElement="wysiwyg">
            <argument name="data" xsi:type="array">
                <item name="config" xsi:type="array">
                    <item name="wysiwygConfigData" xsi:type="array">
                        <item name="height" xsi:type="string">100px</item>
                        <item name="add_variables" xsi:type="boolean">false</item>
                        <item name="add_widgets" xsi:type="boolean">false</item>
                        <item name="add_images" xsi:type="boolean">true</item>
                        <item name="add_directives" xsi:type="boolean">true</item>
                    </item>
                    <item name="source" xsi:type="string">category</item>
                </item>
            </argument>
            <settings>
                <label translate="true">Below Description</label>
                <dataScope>custom_description</dataScope>
            </settings>
            <formElements>
                <wysiwyg class="Magento\Catalog\Ui\Component\Category\Form\Element\Wysiwyg">
                    <settings>
                        <rows>8</rows>
                        <wysiwyg>true</wysiwyg>
                    </settings>
                </wysiwyg>
            </formElements>
        </field>

```

4. Try to edit the field on the category - it won't save. Note in our case our custom fields already have values that display on the frontend but do not show on the backend. Changes do not save,

5. Change the sort order of the field to 49 and try again., This time you can edit and save the custom attribute but not the normal Description field.

6. Change the sort order back to 100 and make the field a basic textarea field e.g.

```
<field name="custom_description" sortOrder="100" formElement="textarea">
    <settings>
        <dataType>string</dataType>
        <label translate="true">My Text Attribute</label>
    </settings>
</field>

```
Now you can edit and save the field as text.

### Expected result

We are able to edit all HTML fields on the category page.,

### Actual result

We can only edit the first HTML field that appears on the category page.

### Additional information

Custom attribute editor empty:
![image](https://github.com/magento/magento2/assets/13659750/cc2732d7-48d5-4622-9410-519d74b62b70)

Move custom attribute before core description, now description empty:
![image](https://github.com/magento/magento2/assets/13659750/a9acc282-967f-4d3b-90b9-64cf1c9134a4)

Make custom attribute a standard textarea and works ok:
![image](https://github.com/magento/magento2/assets/13659750/7e54987c-489f-4858-8046-2cd05d30ec20)

### Release note

Fix bug introduced in 2.4.? that prevents editing of additional custom HTML attributes on categories.

### Triage and priority

- [X] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by adamlavery** [comment]
@magento give me 2.4-develop instance

**Comment by adamlavery** [comment]
How do we get SSH access to this to be able to add a test module?

**Comment by GamesmenJordan** [comment]
+1 to this issue.

**Comment by engcom-Hotel** [comment]
Hello @adamlavery,

Thanks for the report and collaboration!

We have tried to reproduce the issue by upgrading the Magento from 2.4.3 to 2.4.7-p1 (The latest released version). But it seems the issue is not reproducible for us. After upgrading we can see the saved value from the previous version i.e. 2.4.3.

Please refer to the below screenshot  for reference:

![image](https://github.com/magento/magento2/assets/51681618/b93407dd-0a0b-45b2-bce0-7582257b3609)

We are also attaching the module we have created to reproduce the issue:
[Magentoissues.zip](https://github.com/user-attachments/files/16127225/Magentoissues.zip)

Let us know in case we missed anything.

Thanks

**Comment by adamlavery** [comment]
@engcom-Hotel this really doesn't demonstrate the issue. Try moving your custom attribute into Content and display after the core Description field (i.e. sortOrder 51+). As per this test, your custom attribute is the first HTML field on the page and that works. I suspect if you open Content you'll find you can't change Description. But do as we have done to exactly reproduce our scenario.

**Comment by engcom-Hotel** [comment]
Hello @adamlavery,

Thanks for the response. 

As per your response, we have made the required changes in the module and have moved the custom category attribute to Display Settings. However, we still cannot reproduce the issue.

We followed the entire upgrade process from 2.4.3 to 2.4.7-p1. Please refer to the screenshot below for reference:

![image](https://github.com/magento/magento2/assets/51681618/8f1dce09-3a4a-42ff-be11-ab59c2765c1a)

Let us know if we have missed anything.

Thanks

**Comment by adamlavery** [comment]
Hi @engcom-Hotel , ok so this time you have put it in the Display Settings section. Put it in the **Content** section, below Description as per the screenshots.

**Comment by engcom-Hotel** [comment]
Hello @adamlavery, @skovalenk,

> It is a shame guys, that you are saying that you could not even reproduce it. it is 100% reproducible everywhere

I apologize but we are trying our best in this issue reproduction. 

We have tried again and now the custom field is in the Content section below the description field. Please refer to the screenshot below for reference:

![image](https://github.com/magento/magento2/assets/51681618/4fb6d229-80a4-477e-beee-459b496e0f8d)

But the issue is not reproducible for us. Just so you know, we are attaching the module as well, please let us know if we missed anything.

[Magentoissues.zip](https://github.com/user-attachments/files/16161714/Magentoissues.zip)

Thanks

**Comment by adamlavery** [comment]
Thanks @skovalenk -  I can confirm that reverting dom-observer.js back to the original version fixes the issue. 

![image](https://github.com/magento/magento2/assets/13659750/556d7ba0-6871-4017-852b-28704e832c40)


I noticed there are many instances of the same kind of change in that script, I guess any one of which could be causing other issues yet to be discovered. Hence I reverted the whole script. I may be wrong but this seems to me a case of someone rewriting this code for the sake of rewriting it the way they prefer to do it. Can't see that the changes add anything - other than new bugs!!

Clearly the automated test scripts are not up to much as they haven't picked up that this "fix" actually creates a new bug.

Too much of this happens in Magento nowdays. Old adage: if it ain't broke don't fix it - changing it might actually break it!!! Whoever made this change needs to review what they did (and why).

**Comment by adamlavery** [comment]
@engcom-Hotel try specfying the field in the same way we have i.e. same XML. Your XML is different. For some reason your label doesn't display. Who know's what effect a slightly different way of adding the field might be having in the black-box that is Magento UI.

That's been one of the big issues of M2 UI - too many different ways to do the same thing. Can't wait to see the whole over-engineered mess stripped out and replaced wth something much simpler - like Hyva.

**Comment by adamlavery** [comment]
@engcom-Hotel another slight difference - your attribute is varchar, ours is text.

**Comment by engcom-Hotel** [comment]
Hello @adamlavery,

We can reproduce the issue now. After the upgrade, we cannot see the values in the custom attribute. Please refer to the below screenshots for reference:

_**If the Custom attribute is below the description field**_

_We can’t  see the value in the custom attribute

![image](https://github.com/magento/magento2/assets/51681618/b11573d6-0b7c-4094-8f60-bafcd81ce18d)

_**If the Custom attribute is above the description field**_

_We can see the value in the custom attribute

![image](https://github.com/magento/magento2/assets/51681618/b009741d-ed93-452a-9d05-af7e1dd54ef8)

Hence confirming the issue.

Thanks

**Comment by hostep** [comment]
I have the feeling this might have already been fixed in 2.4-develop with https://github.com/magento/magento2/pull/38623, but I might be wrong.

@engcom-Hotel: did you only test it while upgrading Magento to 2.4.7, or did you then also test it by further upgrading to the latest `2.4-develop` version?

**Comment by adamlavery** [comment]
I can confirm that this commit fixes the issue: [1933e66](https://github.com/magento/magento2/commit/1933e66b2bea830d866dd850fdac22c2164ac22d)

**Comment by hostep** [comment]
Okay perfect, then I guess we can close this issue, the fix is probably scheduled to be delivered in Magento 2.4.8

**Comment by engcom-Hotel** [comment]
> I have the feeling this might have already been fixed in 2.4-develop with #38623, but I might be wrong.
> 
> @engcom-Hotel: did you only test it while upgrading Magento to 2.4.7, or did you then also test it by further upgrading to the latest `2.4-develop` version?

Hello @hostep,

We have tried it in both  2.4-develop (the latest development branch) and via upgrading. The issue is not reproducible in 2.4-develop but it is reproducible in upgrading from 2.4.3 to 2.4.7-p1 (the latest released version). 

So as per my suggestion, the issue should remain open, for upgrading scenario.

Thanks

**Comment by adamlavery** [comment]
@hostep - agreed! This has nothing to do with upgrading from one release to another. It's a coding error introduced by https://github.com/magento/magento2/pull/37477.

Is it not considered serious enough to go in the next patch release? How can we know which release includes it?

**Comment by hostep** [comment]
@engcom-Hotel: ticket can be closed in my opinion, it's a bug in the code that has introduced the problem, not the upgrade process itself. The reason why it's not reproducible in 2.4-develop is because the fix was already introduced in 2.4-develop on [28 May 2024](https://github.com/magento/magento2/pull/38623#event-12959248327)

@adamlavery: I assume this fix will get included in Magento 2.4.8 and not in one of the security patch releases. I don't think Adobe considers it as highly critical.

Just patch your codebase to fix the bug if you need it urgently, using one of the well known composer patch libraries (https://github.com/vaimo/composer-patches or https://github.com/cweagans/composer-patches/ or ...).

**Comment by engcom-Hotel** [comment]
Sure @hostep. We are closing the issue.

Thanks
